add engine requirements to `create-turbo/package.json`, fix `examples/with-pnpm`

charkour opened the pull request:
Hey, didn't know what was best for this, but I thought this PR could be a good starting point. 

Currently, the docs do not list out what node version or npm version is necessary to run turborepo and some users have run into issues with it: #184 

This PR adds the engine versions to the base package.json for the template in `create-turbo` and also updates the documentation to specify which versions of node and npm are needed. Adds an `.npmrc` to enforce the npm version if using npm. I tested different combinations of npm and node versions and concluded turborepo works with node >= v14 and npm >= v7. I didn't try different pnpm versions, but could test that as well if you would like. 

Also, maybe it would be good to have `create-turbo` and the `turbo` cli check the node and npm version and display an error when the user trys to run commands. If that is a good idea, that is something else this PR could include. As an aside, building and running `create-turbo` locally works fine but jest was presenting errors. 

Thanks for your time!

jaredpalmer commented on docs/pages/docs/getting-started.mdx:
```diff
@@ -23,7 +23,7 @@ To see more examples and starters, have a look at the [example directory on GitH
 Turborepo was designed to be incrementally adopted. Adding it to an existing monorepo takes just a few minutes.
 
-Turborepo works with [Yarn v1](https://classic.yarnpkg.com/lang/en/), [NPM](https://npmjs.com), and [PNPM](https://pnpm.io/) workspaces. The `turbo` CLI works on the following operating systems.
+Turborepo works with [Yarn v1](https://classic.yarnpkg.com/lang/en/), [NPM](https://npmjs.com), and [PNPM](https://pnpm.io/) workspaces and [Node](https://nodejs.org/en/) v14 or higher. The `turbo` CLI works on the following operating systems.
```
Turbo is written in Go and so this is not entirely accurate. `turbo` doesn't even need Node.js installed to function. 

`create-turbo` likely needs Node 12 or 14 to work is my guess, and the starter examples likely need the same. Choosing NPM package manager also would require npm > 7 (the first to support workpsaces).

I think this is too much information for "getting started" so I would rather enforce it in the NPM `package.json`

charkour replied:
Ah right, thanks for that clarification. That makes sense. The project created by `create-turbo` needs Node 14 to function based on my testing of running `npx create-turbo@latest` on different Node versions and then running `npm run build` in the turbopreo. 

I'll update this PR.

charkour commented:
Hey, thanks for taking the time to review this. 

I have removed the Node and NPM version requirements from the docs since it is true `turbo` doesn't depend on Node. Adding the engine requirements to `package.json` in the project bootstrapped from `create-turbo` clarifies the requirements for that bootstrapped project. 

Also, I add the minimum PNPM version. PNPM [added workspaces in v3.7.0](https://pnpm.io/workspaces#workspace-protocol-workspace) according to their docs which I verified using the `examples/with-pnpm` project and installing different PNPM versions. The `with-pnpm` project was missing scripts which were defined in `with-pnpm/readme.md` and added needed turbo `dev` configuration. 

Thanks!

jaredpalmer commented:
Thanks!
